### PR TITLE
[NATS-Streaming] Fix issue in acknowledging messages in auto-ack mode

### PR DESCRIPTION
**Pull request by aashikam**
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/20516

## Purpose
$title 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/messaging/nats/src/main/java/org/ballerinalang/nats/streaming/consumer/Attach.java [hunk 1]
```diff
@@ -41,8 +42,15 @@ public static void streamingAttach(ObjectValue streamingListener, ObjectValue se
         ConcurrentHashMap<ObjectValue, StreamingListener> serviceListenerMap =
                 (ConcurrentHashMap<ObjectValue, StreamingListener>) streamingListener
                         .getNativeData(STREAMING_DISPATCHER_LIST);
-        serviceListenerMap.put(service, new StreamingListener(service, BRuntime.getCurrentRuntime(),
-                                                              streamingListener.getObjectValue("connection")
-                                                                      .getStringValue(Constants.URL)));
+        boolean manualAck = getAckMode(service);
+        serviceListenerMap.put(service, new StreamingListener(service, manualAck, BRuntime.getCurrentRuntime(),
+                streamingListener.getObjectValue("connection")
+                        .getStringValue(Constants.URL)));
```
Comment (kalaiyarasiganeshalingam) [inline stdlib/messaging/nats/src/main/java/org/ballerinalang/nats/streaming/consumer/Attach.java]: Shall we combine this line?

File: stdlib/messaging/nats/src/main/java/org/ballerinalang/nats/streaming/consumer/Attach.java [hunk 2]
```diff
@@ -41,8 +42,15 @@ public static void streamingAttach(ObjectValue streamingListener, ObjectValue se
         ConcurrentHashMap<ObjectValue, StreamingListener> serviceListenerMap =
                 (ConcurrentHashMap<ObjectValue, StreamingListener>) streamingListener
                         .getNativeData(STREAMING_DISPATCHER_LIST);
-        serviceListenerMap.put(service, new StreamingListener(service, BRuntime.getCurrentRuntime(),
-                                                              streamingListener.getObjectValue("connection")
-                                                                      .getStringValue(Constants.URL)));
+        boolean manualAck = getAckMode(service);
+        serviceListenerMap.put(service, new StreamingListener(service, manualAck, BRuntime.getCurrentRuntime(),
```
Comment (wggihan) [inline stdlib/messaging/nats/src/main/java/org/ballerinalang/nats/streaming/consumer/Attach.java]: This is not formatted according to the standard. Previous was correct.

File: stdlib/messaging/nats/src/main/java/org/ballerinalang/nats/streaming/consumer/Attach.java (hunk 1 above)
Comment (aashikam) [reply stdlib/messaging/nats/src/main/java/org/ballerinalang/nats/streaming/consumer/Attach.java]: https://github.com/ballerina-platform/ballerina-lang/pull/20658#discussion_r367913690

File: stdlib/messaging/nats/src/main/java/org/ballerinalang/nats/streaming/consumer/Attach.java (hunk 2 above)
Comment (aashikam) [reply stdlib/messaging/nats/src/main/java/org/ballerinalang/nats/streaming/consumer/Attach.java]: Fixed this in 128133cde87cddcd31fc07a47ca7dada6e83fed0

**Pull request merged**